File: WebAPP/ClientApp/src/dto/stayTeam.ts
import { Stay } from "./stay";

export class StayTeam {
    public id: number;
    public stayId: number;
    public stay: Stay;
    public partnerName: string;

    constructor(id?: number, stayId?: number, stay?: Stay, partnerName?: string)

    constructor(id: number, stayId: number, stay: Stay, partnerName: string) {
        this.id = id;
        this.stayId = stayId;
        this.stay = stay;
        this.partnerName = partnerName;
    }
}

File: Model/StayTeam.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Model
{
    [Table("stay_team")]
    public class StayTeam
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public int? Id { set; get; }
        [Column("stay_id")]
        public int? StayId { set; get; }
        public Stay Stay { set; get; }

        [Column("partner_name")]
        public string PartnerName { set; get; }
    }
}
